File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionOrderRepository.java
package com.eager.questioncloud.core.domain.payment.infrastructure.repository;

import com.eager.questioncloud.core.domain.payment.model.QuestionOrder;
import java.util.List;

public interface QuestionOrderRepository {
    Boolean isAlreadyPurchased(Long userId, List<Long> questionIds);

    void save(QuestionOrder questionOrder);

    void deleteAllInBatch();
}

File: qc-core/qc-pay/qc-pay-core/src/main/kotlin/com/eager/questioncloud/payment/repository/QuestionOrderRepository.kt
package com.eager.questioncloud.payment.repository

import com.eager.questioncloud.payment.domain.QuestionOrder
import com.eager.questioncloud.payment.dto.QuestionOrderData

interface QuestionOrderRepository {
    fun save(questionOrder: QuestionOrder)
    
    fun deleteAllInBatch()
    
    fun getQuestionOrderData(orderId: String): List<QuestionOrderData>
}
